mllib/src/main/scala/org/apache/spark/ml/ann/Layer.scala

```diff
@@ -532,8 +427,13 @@ private[ml] object FeedForwardTopology {
     for(i <- 0 until layerSizes.length - 1){
       layers(i * 2) = new AffineLayer(layerSizes(i), layerSizes(i + 1))
       layers(i * 2 + 1) =
-        if (softmax && i == layerSizes.length - 2) {
-          new FunctionalLayer(new SoftmaxFunction())
+        if (i == layerSizes.length - 2) {
+          if (softmax) {
```

Comment:
nit: maybe instead of `softmax`, have an `isBinaryClassification` flag to make intention clear